### PR TITLE
Add pennylane-lightning-kokkos recipe.

### PR DESCRIPTION
**Pull request by vincentmr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by vincentmr** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by vincentmr** [comment]
https://github.com/orgs/conda-forge/teams/help-python-c, ready for review! (Sorry for prompting twice, I just realized I had tagged the wrong team for review.)

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by vincentmr** [comment]
Thanks a lot @carterbox for this review.

File: recipes/pennylane-lightning-kokkos/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  url: https://github.com/PennyLaneAI/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: f51ba7718defc7bb5064f690f381e04b2ec58cb09f22a171ae5f410860716e30
+  patches:
+    - 0001-setup.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  script: {{ PYTHON }} -m pip install . --no-deps -vv  # [linux]
+  # script: {{ PYTHON }} setup.py build_ext --define="PLKOKKOS_ENABLE_WARNINGS=OFF" && {{ PYTHON }} setup.py bdist_wheel && {{ PYTHON }} -m pip install . --no-deps -vv  # [osx]
```
[the remark this answers is not in the document]
Comment (loriab) [reply recipes/pennylane-lightning-kokkos/meta.yaml]: On the faint chance that a translation table can be of help for getting osx going, https://github.com/dgasmith/gau2grid/blob/master/gau2grid/c_pragma.py#L67

[the remark this answers is not in the document]
Comment (loriab) [reply recipes/pennylane-lightning-kokkos/meta.yaml]: I think in your cpp file where you're calling `aligned_alloc` that gets replaced by `ALIGNED_MALLOC` that it pulls alternate definitions from the header depending on OS. So one more line in your patch.

[the remark this answers is not in the document]
Comment (vincentmr) [reply recipes/pennylane-lightning-kokkos/meta.yaml]: So I think the problem is when I `#include <memory>`. Wouldn't I need to patch `$BUILD_PREFIX/include/c++/v1/new`, where the faulty `aligned_alloc` is used?

[the remark this answers is not in the document]
Comment (loriab) [reply recipes/pennylane-lightning-kokkos/meta.yaml]: I'm sorry, I thought the `aligned_alloc` was in _your_ code. If the `aligned_alloc` is in the sysroot, then that's out-of-scope to patch -- something is amiss in the build that I have no hints on. Sorry to lead you down a bad path.

One side observation -- I think by switching the order of Python and pybind11 detection https://github.com/PennyLaneAI/pennylane-lightning-kokkos/blob/main/CMakeLists.txt#L131-L139, you can suppress the double-detection of Python here https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=693154&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1285 . The PythonInterp/PythonLibs pair is old, faulty python detection, while the "Found Python" on line 1289 is the new style.

[the remark this answers is not in the document]
Comment (vincentmr) [reply recipes/pennylane-lightning-kokkos/meta.yaml]: > I'm sorry, I thought the `aligned_alloc` was in _your_ code. If the `aligned_alloc` is in the sysroot, then that's out-of-scope to patch -- something is amiss in the build that I have no hints on. Sorry to lead you down a bad path.

No problem, thanks for your suggestions. I'll keep them in my bag for the future.

[the remark this answers is not in the document]
Comment (vincentmr) [reply recipes/pennylane-lightning-kokkos/meta.yaml]: Ok, so the problem was the same as mentioned by @leofang [here](https://github.com/conda-forge/pennylane-feedstock/pull/27#issuecomment-1504524550). I think this is ready now @carterbox

**Comment by carterbox** [comment]
@conda-forge/help-c-cpp, please review

File: recipes/pennylane-lightning-kokkos/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - ninja
+  host:
+    - libgomp      # [linux]
+    - llvm-openmp  # [osx]
+    - kokkos =3.7
+    - pip
+    - pybind11
+    - python
+  run:
+    - pennylane
```
Comment (carterbox) [inline recipes/pennylane-lightning-kokkos/meta.yaml:38]:
```suggestion
    - pennylane >=0.28
```
https://github.com/PennyLaneAI/pennylane-lightning-kokkos/blob/88d5657159a6fb177b6c4dfec66963201bff47ad/setup.py#LL143C20-L143C20

File: recipes/pennylane-lightning-kokkos/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - 0001-setup.patch
+    - 0002-setup.patch  # [osx]
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: |
+    {{ PYTHON }} setup.py build_ext --define="PLKOKKOS_ENABLE_WARNINGS=OFF"
+    {{ PYTHON }} setup.py bdist_wheel
+    {{ PYTHON }} -m pip install . --no-deps -vv
```
Comment (carterbox) [inline recipes/pennylane-lightning-kokkos/meta.yaml:22]: This package installs some unused CMake files into site-packages. Open a patch upstream to remove them for the next release, please.

**Review by carterbox — APPROVED** [review]

File: recipes/pennylane-lightning-kokkos/meta.yaml (hunk 3 above)
Comment (vincentmr) [reply recipes/pennylane-lightning-kokkos/meta.yaml:22]: I see that. The goal is to ship the header templated library along with the binaries. I'll patch the glob expression to fix this, if that alright.

Comment (carterbox) [reply recipes/pennylane-lightning-kokkos/meta.yaml:22]: I think you're asking if modifying `MANIFEST.in` is the correct way to exclude that file. I think it is.

**Pull request merged**